manifests/secondarydns.yaml

```diff
@@ -178,3 +178,60 @@ spec:
           name: secondary-dns
       - name: secdns-zones
         emptyDir: {}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: ksd-allow-egress-to-dns
+  namespace: secondary
+spec:
+  podSelector:
+    matchLabels:
+      k8s-app: secondary-dns
+  policyTypes:
+  - Egress
+  egress:
+  - to:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: kube-system
+      podSelector:
+        matchLabels:
+          k8s-app: kube-dns
+    ports:
+    - protocol: TCP
+      port: dns-tcp
+    - protocol: UDP
+      port: dns
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: ksd-allow-egress-to-api-server
+  namespace: secondary
+spec:
+  podSelector:
+    matchLabels:
+      k8s-app: secondary-dns
+  policyTypes:
+  - Egress
+  egress:
+  - ports:
+    - protocol: TCP
+      port: 6443
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-ingress-to-secondary-dns
+  namespace: secondary
+spec:
+  podSelector:
+    matchLabels:
+      k8s-app: secondary-dns
+  policyTypes:
+  - Ingress
+  ingress:
+  - ports:
+    - protocol: UDP
+      port: 5353
```

Comment:
how do you know this is the port you're listening to ? 

Which type of "app" is this ? A controller (why do you need to allow ingress to it ?), a webhook, something else ?... 

Pretty much after "what's listening behind this port and why".

Reply:
https://github.com/kubevirt/kubesecondarydns/blob/51a9d78d5d810dd7be8aec3666c9afa8001871cf/manifests/secondarydns.yaml#L12

is it dedicated coreDNS that we put as container in the KSD deployment 

KSD core logic updates the zone file for that container (both are in the same pod, so via shared mount)
this is how we serve DNS requests for secondary interfaces